src/librustc_mir/monomorphize/partitioning.rs

```diff
@@ -487,6 +373,176 @@ fn place_root_mono_items<'a, 'tcx, I>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     }
 }
 
+fn mono_item_linkage_and_visibility(
+    tcx: TyCtxt<'a, 'tcx, 'tcx>,
+    mono_item: &MonoItem<'tcx>,
+    can_be_internalized: &mut bool,
+    default: &dyn Fn(DefId, bool) -> Visibility,
```

Comment:
It seems like `default` could be just a regular function and doesn't need to be passed in as a closure.